app/controllers/questions_controller.rb

```diff
@@ -0,0 +1,41 @@
+class QuestionsController < ApplicationController
+  before_action :find_test
+
+  rescue_from ActiveRecord::RecordNotFound, with: :rescue_with_question_not_found
+
+  def index
+    @questions = Question.where(test: @test)
```

Comment:
```suggestion
    @questions = @test.questions
```
в большинстве случаев правильно использовать ассоциации